packages/components/src/components/eventsCookieConsent/useConsentSiteSettings.tsx

```diff
@@ -0,0 +1,250 @@
+import React from 'react';
+import { DEFAULT_LANGUAGE } from '../../constants';
+import { useCookieConfigurationContext } from '../../cookieConfigurationProvider';
+import useConsentTranslation from '../../hooks/useConsentTranslation';
+import { consentGroupIds } from './constants';
+import type { SiteSettingsLanguage } from './types';
+import { createLocaleDictBuilder } from './utils';
+
+function useLanguages(): SiteSettingsLanguage[] {
+  return [
+    { code: 'en', title: 'English', direction: 'ltr' },
+    { code: 'fi', title: 'Suomi', direction: 'ltr' },
+    { code: 'sv', title: 'Svenska', direction: 'ltr' },
+  ];
+}
+
+function useCookieName() {
+  const { cookieDomain } = useCookieConfigurationContext();
+  const prefix = cookieDomain.replaceAll('.', '-');
+  return `${prefix}-cookie-consents`;
+}
+
+function useConsentContentSource() {
+  const { t } = useConsentTranslation();
+  const languages = useLanguages();
+  const languageCodes = languages.map((lang) => lang.code);
+
+  const buildDict = createLocaleDictBuilder(languageCodes, t);
+  const cookieName = useCookieName();
+  const { cookieDomain, appName } = useCookieConfigurationContext();
+
+  const requiredGroups = [
+    {
+      groupId: consentGroupIds.essentials,
+      title: buildDict('consent:groups.essentials.title'),
+      description: buildDict('consent:groups.essentials.description'),
+      cookies: [
+        {
+          name: cookieName,
+          host: cookieDomain,
+          storageType: 1,
+          description: buildDict('consent:cookies.app', { appName }),
+          expiration: buildDict('consent:expiration.session'),
+        },
+        {
+          name: 'wordpress_*, wp-settings-*',
+          host: 'api.hel.fi',
+          storageType: 1,
+          description: buildDict('consent:cookies.wordpress'),
+          expiration: buildDict('consent:expiration.session'),
+        },
+        {
+          name: 'linkedevents-api-prod-csrftoken',
+          host: 'api.hel.fi',
+          storageType: 1,
+          description: buildDict('consent:cookies.linkedevents'),
+          expiration: buildDict('consent:expiration.year'),
+        },
+        {
+          name: 'i18next',
+          host: 'api.hel.fi',
+          storageType: 1,
+          description: buildDict('consent:cookies.i18next'),
+          expiration: buildDict('consent:expiration.session'),
+        },
+      ],
+    },
+  ] as const;
+
+  const optionalGroups = [
+    {
+      groupId: consentGroupIds.serviceMap,
+      title: buildDict('consent:groups.optionalServiceMap.title'),
+      description: buildDict('consent:groups.optionalServiceMap.text'),
+      cookies: [
+        {
+          name: '_pk.fi*',
+          host: 'palvelukartta.hel.fi',
+          storageType: 1,
+          description: buildDict('consent:cookies.servicemap_analytics'),
+          expiration: buildDict('consent:expiration.days', { days: 393 }),
+        },
+        {
+          name: 'ADRUM',
```

Comment:
Did these names etc come from some UI tool you used? Or how did these came to be? e.g. `ADRUM` is a bit inscrutable at a glance. Mostly just curious.